### PR TITLE
operator: Add method to get authenticated from GCP

### PR DESCRIPTION
**Pull request by sasagarw**
**What this PR does / why we need it**:
There isn't an authentication mechanism in the provided Loki configuration. The Loki GCS authentication is handled via the GCS SDK which is typically done with environment variables(For example:GOOGLE_APPLICATION_CREDENTIALS).

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated
- [x] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Comment by sasagarw** [comment]
cc @periklis

File: operator/internal/manifests/compactor.go [hunk 1]
```diff
@@ -99,6 +100,26 @@ func NewCompactorStatefulSet(opts Options) *appsv1.StatefulSet {
...
+			Name:      opts.Stack.Storage.Secret.Name,
+			ReadOnly:  false,
+			MountPath: secretDirectory,
+		})
+		podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, corev1.EnvVar{
+			Name:  EnvGoogleApplicationCredentials,
+			Value: path.Join(secretDirectory, GCSFileName),
+		})
+	}
+
```
Comment (Red-GV) [inline operator/internal/manifests/compactor.go:122]: Maybe this should just be a callable function in `var.go`? Similar to how we do the modifications for the service monitor?

**Review by Red-GV — COMMENTED** [review]
Looks overall fine to me. A small comment though.

File: operator/internal/manifests/var.go [hunk 2]
```diff
@@ -273,3 +279,23 @@ func lokiReadinessProbe() *corev1.Probe {
...
+	})
+	podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      secretName,
+		ReadOnly:  false,
+		MountPath: secretDirectory,
+	})
+	podSpec.Containers[0].Env = append(podSpec.Containers[0].Env, corev1.EnvVar{
+		Name:  EnvGoogleApplicationCredentials,
+		Value: path.Join(secretDirectory, GCSFileName),
+	})
```
Comment (periklis) [inline operator/internal/manifests/var.go:300]: I suggest to continue using `mergo.Merge` for mutating manifests as we do in other places, e.g. [service monitor PKI](https://github.com/grafana/loki/blob/main/operator/internal/manifests/service_monitor.go#L136).

In addition I suggest we move the whole function out of `var.go` to `storage/configure.go` file and follow-up the configure pattern as in `gateway_tenants.go`, i.e.
- First in `storage/configure.go` you place the `ensureCredentialsFor GCS`
- Second in `manifests/object_storage.go` you place the switch based on `opts.Stack.Storage.Secret.Type`
- Third you call the `configureDeploymentForStorageType` and `configureStatefulSetForStorageType` only in the Builder functions (e.g. `BuildQuerier` w/o nesting if blocks in the original constructor functions (e.g. `NewQuerierDeployment`)

This ensures that we have dedicated configure/mutate functions with proper error handling (thanks to `mergo.Merge` while keeping our manifest constructors are pure and minimal as possible.

e.g.
```go
func configureDeploymentForStorageType(d *appsv1.Deployment, t lokiv1beta1.ObjectStorageType) error {
	switch t {
	case lokiv1beta1.ObjectStorageSecretGCS:
		return storage.ConfigureDeployment(d)
	default:
		return nil
	}
}
func configureStatefulSetForStorageType(d *appsv1.Deployment, t lokiv1beta1.ObjectStorageType) error {
	switch t {
	case lokiv1beta1.ObjectStorageSecretGCS:
		return storage.ConfigureStatefulSet(d)
	default:
		return nil
	}
}
```

**Review by periklis — CHANGES_REQUESTED** [review]

**Review by periklis — APPROVED** [review]
Looks really good to me.

**Pull request merged**